### PR TITLE
Resolved out of bound access of shared buffer issue.

### PR DESCRIPTION
**Pull request by quic-anankulk**
While packing shared buffer with names of all shared objects present in custom DSP_LIBRARY_PATH , there is a possibility of buffer overflow if the shared object names are exceeding the desired limit. The change makes sure the limit is not exceeded thus avoiding buffer overflow. Also, the buffer was allocated with 1KB memory which might fall short to accommodate all the needed shared object names so, increasing this size to 2KB.

### CODE REVIEW

File: src/fastrpc_procbuf.c [hunk 1]
```diff
@@ -238,8 +246,16 @@ void fastrpc_process_pack_params(int dev, int domain) {
...
+			nErr = pack_sharedbuf_params(CUSTOM_DSP_SEARCH_PATH_LIBS_ID,
+					lib_names, buffer_size, start_addr, cur_addr);
+			if (nErr) {
+				FARF(ERROR, "Error 0x%x: %s: Failed to pack the directory list in shared buffer",
+						nErr, __func__);
+			}
+		}
 	}
 	nErr = pack_proc_shared_buf_params(domain, CUSTOM_DSP_SEARCH_PATH_LIBS_ID,
 			lib_names, buffer_size);
```
Comment (ekanshibu) [inline src/fastrpc_procbuf.c]: if malloc fails, NULL lib_names will get passed here, is that handled properly?

File: src/fastrpc_procbuf.c [hunk 2]
```diff
@@ -69,13 +70,15 @@ int proc_sharedbuf_init(int dev, int domain) {
 
 static int get_non_preload_lib_names (char** lib_names, size_t* buffer_size, int domain)
 {
-	int nErr = AEE_SUCCESS, env_list_len = 0;
-	char* data_paths = NULL;
-	char *saveptr       = NULL;
+	int nErr = AEE_SUCCESS, env_list_len = 0, concat_len = 0;
+	int new_len = 0, additional_bytes = 2;
+	char *data_paths = NULL, *saveptr       = NULL;
```
Comment (ekanshibu) [inline src/fastrpc_procbuf.c]: too much space in saveptr declaration

Comment (quic-anankulk) [reply src/fastrpc_procbuf.c]: Addressed the comment.

File: src/fastrpc_procbuf.c (hunk 1 above)
Comment (quic-anankulk) [reply src/fastrpc_procbuf.c]: Addressed the comment.

File: src/fastrpc_procbuf.c [hunk 3]
```diff
@@ -85,23 +88,28 @@ static int get_non_preload_lib_names (char** lib_names, size_t* buffer_size, int
...
-					std_strlcat(*lib_names, ";", new_len);
-					*buffer_size = new_len;
+				if (std_strstr(file, FILE_EXT) != NULL) {
+					if (concat_len + std_strlen(file) > MAX_NON_PRELOAD_LIBS_LEN) {
+						FARF(ALWAYS,"ERROR: Failed to pack library names in custom DSP_LIBRARY_PATH as required buffer size exceeds Max limit (%d).", MAX_NON_PRELOAD_LIBS_LEN);
+						nErr = AEE_EBUFFERTOOSMALL;
+						goto bail;
+					}
+					std_strlcat(*lib_names, file, MAX_NON_PRELOAD_LIBS_LEN);
+					concat_len = std_strlcat(*lib_names, ";", MAX_NON_PRELOAD_LIBS_LEN);
```
Comment (ekanshibu) [inline src/fastrpc_procbuf.c:98]: should this be concat_len += std_strlcat(*lib_names, ";", MAX_NON_PRELOAD_LIBS_LEN);

**Comment by quic-ekangupt** [comment]
Can you create the branch off the "development" branch of this repo and raise the PR?
https://github.com/quic/fastrpc/blob/development/CONTRIBUTING.md

**Pull request closed without merging**